Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -897,6 +935,17 @@ TextureCache::TCacheEntryBase* TextureCache::AllocateTexture(const TCacheEntryCo
 	return g_texture_cache->CreateTexture(config);
 }
 
+TextureCache::TexCache::iterator TextureCache::RemoveTextureFromCache(TexCache::iterator iter)
+{
+	if (iter->second->textures_by_hash_iter != textures_by_address.end())
+	{
+		textures_by_hash.erase(iter->second->textures_by_hash_iter);
```

Comment:
I'd also clear textures_by_hash_iter here, just to be sure it's always cleared for new textures.